### ISSUE TITLE
Kit with preferredGenerator isn't finding generator

### ISSUE THREAD

**Issue by gcampbell-msft** [microsoft/vscode-cmake-tools#4031]
Reproduce issue like this:
Install jom (https://wiki.qt.io/Jom).
Add jom.exe location to your system PATH.
In cmake-tools-kits.json add the following kit:
`  {"name": "JOM test kit", "preferredGenerator": {"name": "NMake Makefiles JOM"}}`

Open a CMake project and try to configure it with the "JOM test kit".
You will get the error:

> [main] Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json, your Kit configuration or PATH variable. Error: No usable generator found.

_Originally posted by @KorkiFork in https://github.com/microsoft/vscode-cmake-tools/issues/4019#issuecomment-2318895439_

**Comment by gcampbell-msft** [comment]
Ah, it might be that the PR from #4028 fixes this? I'll try to reproduce your project, despite not having a full repro. Thanks

**Comment by gcampbell-msft** [comment]
@marcogmaia Could you test with the vsix from this comment: https://github.com/microsoft/vscode-cmake-tools/issues/4023#issuecomment-2319719783?

**Comment by marcogmaia** [comment]
Still not working.

Diagnostic log:

```json
{
  "os": "win32",
  "vscodeVersion": "1.92.2",
  "cmtVersion": "1.13.0",
  "configurations": [
    {
      "folder": "c:\\Users\\marco\\Documents\\dev\\random_folder",
      "cmakeVersion": "3.30.0",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": true,
      "compilers": {
        "C": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe",
        "CXX": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": false
    }
  ]
}
```

Build log:
```log
vcpkg_installed -DVCPKG_OVERLAY_TRIPLETS=C:/Users/marco/Documents/dev/random_folder/3rd_party/triplets -DCMAKE_TOOLCHAIN_FILE=C:/Users/marco/Documents/dev/games/git/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_INSTALL_PREFIX=C:/Users/marco/Documents/dev/random_folder/out/install/windows-debug -SC:/Users/marco/Documents/dev/random_folder -BC:/Users/marco/Documents/dev/random_folder/out/build/windows-debug -G Ninja
[cmake] -- Running vcpkg install
[cmake] All requested packages are currently installed.
[cmake] Total install time: 200 ns
[cmake] bext-di is header-only and can be used from CMake via:
...
...
[cmake]   find_package(Qt6Xml CONFIG REQUIRED)
[cmake]   target_link_libraries(main PRIVATE Qt::Xml Qt6::Xml Qt::XmlPrivate Qt6::XmlPrivate)
...
...
[cmake] -- Running vcpkg install - done
[cmake] -- The C compiler identification is unknown
[cmake] -- The CXX compiler identification is unknown
[cmake] CMake Error at CMakeLists.txt:21 (project):
[cmake]   The CMAKE_C_COMPILER:
[cmake] 
[cmake]     cl.exe
[cmake] 
[cmake]   is not a full path and was not found in the PATH.  Perhaps the extension is
[cmake]   missing?
[cmake] 
[cmake]   Tell CMake where to find the compiler by setting either the environment
[cmake]   variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
[cmake]   the compiler, or to the compiler name if it is in the PATH.
[cmake] 
[cmake] 
[cmake] CMake Error at CMakeLists.txt:21 (project):
[cmake]   The CMAKE_CXX_COMPILER:
[cmake] 
[cmake]     cl.exe
[cmake] 
[cmake]   is not a full path and was not found in the PATH.  Perhaps the extension is
[cmake]   missing?
[cmake] 
[cmake]   Tell CMake where to find the compiler by setting either the environment
[cmake]   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
[cmake]   to the compiler, or to the compiler name if it is in the PATH.
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=cl.exe -DCMAKE_CXX_COMPILER=cl.exe -DX_VCPKG_APPLOCAL_DEPS_INSTALL=1 -DVCPKG_INSTALLED_DIR=C:/Users/marco/Documents/dev/random_folder/out/vcpkg_installed -DVCPKG_OVERLAY_TRIPLETS=C:/Users/marco/Documents/dev/random_folder/3rd_party/triplets -DCMAKE_TOOLCHAIN_FILE=C:/Users/marco/Documents/dev/games/git/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_INSTALL_PREFIX=C:/Users/marco/Documents/dev/random_folder/out/install/windows-debug -SC:/Users/marco/Documents/dev/random_folder -BC:/Users/marco/Documents/dev/random_folder/out/build/windows-debug -G Ninja exited with code: 1
```

**Comment by gcampbell-msft** [comment]
Could you please make a minimal repro so that I can directly test so that I can best investigate?

**Comment by gcampbell-msft** [comment]
@marcogmaia I've tested that this configure preset works as expected: 

{
    "version": 8,
    "configurePresets": [
        {
            "name": "base",
            "toolchainFile": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
            "generator": "Ninja"
        },
        {
            "name": "windows",
            "inherits": "base",
            "cacheVariables": {
                "CMAKE_C_COMPILER": "cl.exe",
                "CMAKE_CXX_COMPILER": "cl.exe"
            }
        }
    ]
}

**Comment by marcogmaia** [comment]
I'll test it later with a clean repo to see if it works with a minimal setup.

**Comment by gcampbell-msft** [comment]
@marcogmaia Thank you, apologies for the confusion, we're working on a couple of different items that slightly regressed, I appreciate your patience and willingness to help test. If you can get me a CMakeLists.txt to go with a the presets you passed that would be helpful. As far as I currently can tell #4028 fixes the issue right now. We hope to merge that PR today sometime

**Comment by gcampbell-msft** [comment]
@KorkiFork Could you test with this vsix, modify the extension from .zip to .vsix? I believe that it may fix your issue and it would help if you could confirm

[cmake-tools.zip](https://github.com/user-attachments/files/16819857/cmake-tools.zip)

**Comment by gcampbell-msft** [comment]
The fix for this issue will be released in the next hour or so in v1.20.2.

**Comment by marcogmaia** [comment]
@gcampbell-msft, the pre-release version is working fine. Thanks.
